### ISSUE TITLE
Clarify that shlex.quote does not quote shell keywords

### ISSUE THREAD

**Issue by 53c06163-9fa6-4ecd-824b-4f9281315044** [python/cpython#70312]
BPO | [26124](https://bugs.python.org/issue26124)
--- | :---
Nosy | @freddrake, @merwok, @bitdancer, @serhiy-storchaka
PRs | <li>python/cpython#13333</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2016-01-15.13:14:41.016>
labels = ['easy', 'type-bug', 'library', 'docs']
title = 'shlex.quote and pipes.quote do not quote shell keywords'
updated_at = <Date 2019-05-15.01:34:47.604>
user = 'https://bugs.python.org/CharlesDaffern'
```

bugs.python.org fields:
```python
activity = <Date 2019-05-15.01:34:47.604>
actor = 'Windson Yang'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Documentation', 'Library (Lib)']
creation = <Date 2016-01-15.13:14:41.016>
creator = 'Charles Daffern'
dependencies = []
files = []
hgrepos = []
issue_num = 26124
keywords = ['patch', 'easy']
message_count = 9.0
messages = ['258292', '258298', '258299', '258300', '258303', '258304', '258318', '258322', '258550']
nosy_count = 6.0
nosy_names = ['fdrake', 'eric.araujo', 'r.david.murray', 'docs@python', 'serhiy.storchaka', 'Charles Daffern']
pr_nums = ['13333']
priority = 'normal'
resolution = 'wont fix'
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue26124'
versions = ['Python 3.6']
```

</p></details>

**Comment by serhiy-storchaka** [comment]
I am not sure that this change is necessary at all. `shlex.quote()` is usually used to quote arguments for shell commands, and shell keywords do not have special meaning here. If you use arbitrary user input as a command, you perhaps have a large security issue.

**Comment by slateny** [comment]
> ... If you use arbitrary user input as a command, you perhaps have a large security issue.

But [the section](https://docs.python.org/dev/library/shlex.html#shlex.quote) has an example on a malicious filename, so it does feel like it implies it can be used for arbitrary user input for security. If that isn't the case, then perhaps the section should be revamped a bit to indicate that.

**Comment by merwok** [comment]
I agree that the behaviour is correct («shell utilities» does not mean «shell-like interpreter» but «toolbox to run command lines»), but we can’t say it is obvious when people do report that they have other expectations.  Adding some clarification to the doc would help!

**Comment by serhiy-storchaka** [comment]
If you add some clarification here, then you need to add also a warning that `python <filename>` is unsafe because it can execute an arbitrary code.

**Comment by merwok** [comment]
You are unnecessarily hostile here.

**Comment by serhiy-storchaka** [comment]
I sincerely apologize. I've gone too far. I will be more careful with my words in the future.

Would you mind to create a patch? I'm afraid that a new contributor might get confused about the problem and what should be changed.

Note that not quoting shell keywords usually does not cause any problems. Adding a special clarification can create a false impression that the user should do something with this (like manually quoting them), while this is not needed. It only causes an issue in the case which we should discourage.

**Comment by merwok** [comment]
No harm done!  Thanks for the words.

I will propose a patch, maybe this weekend.  The original poster won’t see this but I’ll ask review from slateny and core devs who’ve made changes to shutil.

**Comment by encukou** [comment]
IMO, the current docs are quite clear:

> The returned value is a string that can safely be used as one token in a shell command line

I'll close the issue. If you disagree, please reopen (or open a new one; ideally with a PR).
